### PR TITLE
existsSync should never throw ENAMETOOLONG

### PR DESCRIPTION
**Pull request by nektro**
Fixes https://github.com/oven-sh/bun/issues/10356

### CODE REVIEW

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
1) This de-optimizes `fs.exists` and `fs.promises.exists` because `PathLike` has better code paths for handling file paths that may be used across threads.

2) This specifically handles `ENAMETOOLONG`, but a future change may add more errors for us to throw, and it will throw in those cases (leading the original issue to regress again)

I think a better fix here would be to define `existsSync` as something like
```ts
(path: string) => { 
  try {
     return fs.existsSync.$apply(fs, arguments);
  } catch (e) {
     return false;
  } 
}
```


Which is closer to what Node does as well

File: src/js/node/fs.promises.ts [hunk 1]
```diff
@@ -159,7 +159,13 @@ const exports = {
   close: fs.close.bind(fs),
   copyFile: fs.copyFile.bind(fs),
   cp,
-  exists: fs.exists.bind(fs),
+  exists: async function exists() {
+    try {
+      return fs.exists.$apply(fs, arguments);
```
Comment (Jarred-Sumner) [inline src/js/node/fs.promises.ts]: We have to await this value or a rejected Promise will throw

```suggestion
      return await fs.exists.$apply(fs, arguments);
```

File: src/js/node/fs.ts [hunk 2]
```diff
@@ -309,7 +309,13 @@ var access = function access(...args) {
   appendFileSync = fs.appendFileSync.bind(fs),
   closeSync = fs.closeSync.bind(fs),
   copyFileSync = fs.copyFileSync.bind(fs),
-  existsSync = fs.existsSync.bind(fs),
+  existsSync = function existsSync() {
```
Comment (Jarred-Sumner) [inline src/js/node/fs.ts:318]: Don't forget about `fs.exists`, the asynchronous callback-calling kind

https://github.com/nodejs/node/blob/c82f3c9e80f0eeec4ae5b7aedd1183127abda4ad/lib/fs.js#L248-L266

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
2 comments

File: src/js/node/fs.ts (hunk 2 above)
Comment (nektro) [reply src/js/node/fs.ts:318]: that one's already try/catch'd but I'll add a test

Comment (nektro) [reply src/js/node/fs.ts:318]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/fs.d.ts#L3468 maybe I shouldn't?

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**